packages/gatsby-source-kontent/gatsby-node.ts

```diff
@@ -15,6 +15,36 @@ import { handleIncomingWebhook } from './src/webhookProcessor';
 
 let itemTypes: string[];
 
+exports.pluginOptionsSchema = ({ Joi }: { Joi: any }) => {
+  return Joi.object({
+    projectId: Joi.string()
+      .required()
+      .description(`Project ID from "Project settings" -> "API keys".`),
+    languageCodenames: Joi.array().items(Joi.string())
+      .required()
+      .description(`Array of language codenames from "Project settings" -> "Localization" - the first one is considered as the default one`),
+    includeTypes: Joi.boolean()
+      .description(`Include content type nodes to GraphQL model.`)
+      .default(false),
+    includeTaxonomies: Joi.boolean()
+      .description(`Include taxonomy node to GraphQL model.`)
+      .default(false),
+    authorizationKey: Joi.string()
+      .description(`For preview/secured API key - depends on "usePreviewUrl" config`),
+    usePreviewUrl: Joi.boolean()      
+      .description(`When true, "preview-deliver.kontent.ai" used as primary domain for data source.`)
+      .default(false),
+    proxy: Joi.object()
+      .keys({
+        deliveryDomain: Joi.string()
+          .description(`Base url used for all requests. Defaults to "deliver.kontent.ai"`),
+        previewDeliveryDomain: Joi.string()
+          .description(`Base url used for preview requests. Defaults to "preview-deliver.kontent.ai"`)
+      })
+      .description(`Proxy domain setting object`)
+  });
```

Comment:
It is possible to [test out the validity of the keys](https://www.gatsbyjs.com/docs/configuring-usage-with-plugin-options/#validate-external-access), but it requires requests. The number of requests depends on the number of languages. I don't think, it is necessary.

It is also possible to add a new option flag. Something like `validateOptions` and sent the test requests only if this flag is set to `true`. Not sure, what set as a default value though,

Pros:
* More complex error message in the Gatsby log
* Error will appear sooner in case of invalid keys

Cons:
* Raise complexity of code a bit
* Require clean readme about the flag and the behaviour

Reply:
I have decided **not to implement** the [external schema validation](https://www.gatsbyjs.com/docs/configuring-usage-with-plugin-options/#validate-external-access), because it is currently not possible to validate the language codenames provided because even for the nonexisting language will end up HTTP code 200:

```plain
https://deliver.kontent.ai/cbf6934c-956f-0006-799e-42054f50c3a6/items?limit=1&elements=null&language=NONEXISTING
```